code/game/objects/effects/effect_system/particle_effects.dm

```diff
@@ -39,8 +44,8 @@
 	for(var/turf/open/floor/plating/ground/snow/S in loc)//Snow
 		S.fire_act()
 
-/obj/effect/particle_effect/fire/Crossed(mob/living/L)
-	. = ..()
+/obj/effect/particle_effect/fire/proc/on_cross(datum/source, mob/living/L, oldloc, oldlocs)
```

Comment:
General remark, why isn't there a on_cross proc on atom that you override?

This would prevent having to write SIGNAL HANDLER all the time, as well as not having to rewrite the args.

This goes for on_exited as well

Reply:
iirc /atom root procs that are empty are bad but i cant remember why ask spooky

Reply:
Spooky didn't answer me, can you check if this is true?